docs/specification.md

```diff
@@ -1107,6 +2043,155 @@ Once authenticated, the A2A Server authorizes requests based on the authenticate
+
+#### 7.7.1. Data Access and Authorization Scoping
+
+Implementations **MUST** ensure appropriate scope limitation based on the authenticated user's permissions. This applies to all operations that access or list tasks and other user-specific resources.
+
+**Authorization Principles:**
+
+- Servers **MUST** implement authorization checks on every [A2A Protocol Operations](#3-a2a-protocol-operations) request
+- Servers **SHOULD NOT** return tasks, contexts, or data from other users or unauthorized contexts
+- Even when contextId or other filter parameters are not specified in requests, implementations **MUST** scope results to the caller's authorization and tenancy boundaries
+- The implementation **MAY** choose to limit results to:
+    - Tasks created by the current authenticated user
+    - Tasks within a default user context
+    - Return an authorization error if the scope cannot be safely determined
+
+**Operations Requiring Scope Limitation:**
+
+- [`List Tasks`](#314-list-tasks): **MUST** only return tasks visible to the authenticated client
+- [`Get Task`](#313-get-task): **MUST** verify the authenticated client has access to the requested task
+- Task-related operations (Cancel, Resubscribe, Push Notification Config): **MUST** verify task ownership or access rights
+
+**Multi-Tenancy Considerations:**
+
+- Implementations serving multiple tenants **MUST** enforce strict tenant isolation
+- Task IDs, context IDs, and other identifiers **SHOULD** be scoped to prevent cross-tenant access
+- Authorization checks **MUST** occur before any database queries or operations that could leak information about the existence of resources in other tenants
+
+See also: [Section 3.1.4 List Tasks (Security Note)](#314-list-tasks) for operation-specific requirements.
+
+#### 7.7.2. Push Notification Security
+
+When implementing push notifications, both agents (as webhook callers) and clients (as webhook receivers) have security responsibilities.
+
+**Agent (Webhook Caller) Requirements:**
+
+- Agents **MUST** include authentication credentials in webhook requests as specified in [`PushNotificationConfig.authentication`](#433-authenticationinfo)
+- Agents **SHOULD** implement reasonable timeout values for webhook requests (recommended: 10-30 seconds)
+- Agents **SHOULD** implement retry logic with exponential backoff for failed deliveries
+- Agents **MAY** stop attempting delivery after a configured number of consecutive failures
+- Agents **SHOULD** validate webhook URLs to prevent SSRF (Server-Side Request Forgery) attacks:
+    - Reject private IP ranges (127.0.0.0/8, 10.0.0.0/8, 172.16.0.0/12, 192.168.0.0/16)
+    - Reject localhost and link-local addresses
+    - Implement URL allowlists where appropriate
+
+**Client (Webhook Receiver) Requirements:**
+
+- Clients **MUST** validate webhook authenticity using the provided authentication credentials
+- Clients **SHOULD** verify the task ID in the payload matches an expected task they created
```

Comment:
I'm not sure we always expect the webhook receiver to be the creator of the task, perhaps something more generic like:

```
Clients **SHOULD** validate and handle the request in the scope of the taskId / contextId in the payload
```